### ISSUE TITLE
Why won't cuda be enabled as True?

### ISSUE THREAD

**Issue by sangheonEN** [dianwen-ng/Keyword-Spotting-ConvMixer#5]
print("Training using: ", device) -> cpu!!

**- main.py exec print result**
(keyword_spotting_env) D:\STT_project\Keyword-Spotting-ConvMixer> d: && cd d:\STT_project\Keyword-Spotting-ConvMixer && cmd /C "c:\Users\USER\anaconda3\envs\keyword_spotting_env\python.exe c:\Users\USER\.vscode\extensions\ms-python.debugpy-2024.12.0-win32-x64\bundled\libs\debugpy\adapter/../..\debugpy\launcher 3332 -- D:\STT_project\Keyword-Spotting-ConvMixer\main.py 
"
**Training using:  cpu**
---------------Building train dataloader---------------
Using following mask: 25 freq, 25 time
Using mix-up with rate 0.500000
Use dataset mean -8.522 and std 4.052 for normalization
Dataloader performs with noise augmentation using noises from "noise" file. Noise added at 50 dB
NOTE: If desired to change noise decibel, please modify _wav2fbank with desire noise level.
---------------Building evaluation dataloader---------------
Using following mask: 0 freq, 0 time
Using mix-up with rate 0.000000
Use dataset mean -8.522 and std 4.052 for normalization
D:\STT_project\Keyword-Spotting-ConvMixer\main.py:159: FutureWarning: `torch.cuda.amp.GradScaler(args...)` is deprecated. Please use `torch.amp.GradScaler('cuda', args...)` instead.
  scaler = GradScaler()
c:\Users\USER\anaconda3\envs\keyword_spotting_env\lib\site-packages\torch\amp\grad_scaler.py:132: UserWarning: torch.cuda.amp.GradScaler is enabled, but CUDA is not available.  Disabling.
  warnings.warn(
Starting model training ...

=============start training=============
D:\STT_project\Keyword-Spotting-ConvMixer\main.py:53: FutureWarning: `torch.cuda.amp.autocast(args...)` is deprecated. Please use `torch.amp.autocast('cuda', args...)` instead.
  with autocast():
c:\Users\USER\anaconda3\envs\keyword_spotting_env\lib\site-packages\torch\amp\autocast_mode.py:265: UserWarning: User provided device_type of 'cuda', but CUDA is not available. Disabling
  warnings.warn(

**- nvcc -V**
(keyword_spotting_env) D:\STT_project\Keyword-Spotting-ConvMixer>nvcc -V
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2023 NVIDIA Corporation
Built on Wed_Feb__8_05:53:42_Coordinated_Universal_Time_2023
Cuda compilation tools, release 12.1, V12.1.66
Build cuda_12.1.r12.1/compiler.32415258_0

**- pip env**

(keyword_spotting_env) D:\STT_project\Keyword-Spotting-ConvMixer>pip list
Package            Version
------------------ ------------
audioread          3.0.1
certifi            2024.8.30
cffi               1.17.1
charset-normalizer 3.4.0
colorama           0.4.6
decorator          5.1.1
einops             0.3.2
filelock           3.13.1
fsspec             2024.2.0
idna               3.10
importlib_metadata 8.5.0
Jinja2             3.1.3
joblib             1.4.2
lazy_loader        0.4
librosa            0.10.2.post1
llvmlite           0.41.1
MarkupSafe         2.1.5
mpmath             1.3.0
msgpack            1.1.0
networkx           3.0
numba              0.58.1
numpy              1.22.4
packaging          24.2
pip                24.2
pooch              1.8.2
pycparser          2.22
pytorch-ranger     0.1.1
PyYAML             6.0.2
requests           2.32.3
scikit-learn       1.0.2
scipy              1.7.3
setuptools         75.1.0
soundfile          0.12.1
soxr               0.3.7
sympy              1.13.1
threadpoolctl      3.5.0
torch              2.4.0+cu121
torch-optimizer    0.3.0
torchaudio         2.4.0+cu121
tqdm               4.62.3
typing_extensions  4.12.2
urllib3            2.2.3
wheel              0.44.0
zipp               3.20.2

**Comment by sangheonEN** [comment]
if __name__ == '__main__':
    # os.environ['CUDA_VISIBLE_DEVICES']="2"
    -> change "0" 
    os.environ['CUDA_VISIBLE_DEVICES']="0" 
    main()

I fix it!
